src/config.cpp

```diff
@@ -483,12 +483,14 @@ CParseResult CConfig::parseVariable(const std::string& lhs, const std::string& r
 CParseResult CConfig::parseLine(std::string line, bool dynamic) {
     CParseResult result;
 
-    auto         commentPos  = line.find('#');
-    size_t       lastHashPos = 0;
+    line = removeBeginEndSpacesTabs(line);
+
+    auto   commentPos  = line.find('#');
+    size_t lastHashPos = 0;
 
     while (commentPos != std::string::npos) {
         bool escaped = false;
-        if (commentPos < line.length() - 1) {
+        if (commentPos < line.length() - 1 && commentPos != 0) {
```

Comment:
also why not just 
```cpp
if (commentPos == 0)
    return;
```

above the while loop? If commentPos is 0, the line is useless

Reply:
Agree. I never know to which extend I should create PRs. It would also make sense to add

```cpp
if (line.empty())
    return;
```

just below the second space removal because at that time the line is useless aswell. And then remove all `line.empty()` conditions afterwards.

And other thing (probably another PR) is that the parser should probably ignore all whitespaces, not just space and tab, with [std::isspace](https://en.cppreference.com/w/cpp/string/byte/isspace).

Reply:
Done